homeassistant/components/climate/hive.py

```diff
@@ -0,0 +1,151 @@
+"""
+Support for the Hive devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/hive/
+"""
+import logging
+
+from homeassistant.components.climate import (ClimateDevice,
+                                              STATE_AUTO, STATE_HEAT,
+                                              STATE_OFF, STATE_ON)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.components.hive import DATA_HIVE
+
+DEPENDENCIES = ['hive']
+
+_LOGGER = logging.getLogger(__name__)
+HIVE_TO_HASS_STATE = {'SCHEDULE': STATE_AUTO, 'MANUAL': STATE_HEAT,
+                      'ON': STATE_ON, 'OFF': STATE_OFF}
+HASS_TO_HIVE_STATE = {STATE_AUTO: 'SCHEDULE', STATE_HEAT: 'MANUAL',
+                      STATE_ON: 'ON', STATE_OFF: 'OFF'}
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up Hive climate devices."""
+    session = hass.data.get(DATA_HIVE)
+
+    add_devices([HiveClimateEntity(session, discovery_info)])
+
+
+class HiveClimateEntity(ClimateDevice):
+    """Hive Climate Device."""
+
+    def __init__(self, hivesession, hivedevice):
+        """Initialize the Climate device."""
+        self.node_id = hivedevice["Hive_NodeID"]
+        self.node_name = hivedevice["Hive_NodeName"]
+        self.device_type = hivedevice["HA_DeviceType"]
+        self.session = hivesession
+
+        if self.device_type == "Heating":
+            self.session.heating = self.session.core.Heating()

```

Comment:
the idea originally was to create a new instance in the platform when that platform is needed, so for me I don't have any lights, so it would never create a lights instance.
but the sensors need the heating and hotwater instances as well as the climate, but if sensors come up first then they error as the instance was not created in the climate code first.
The new instances for all platforms at the moment are in the first instance created in the component file.
I can remove from all the platform files and leave only in the component file

Reply:
Yeah, don't duplicate logic if not needed. It's a bad design pattern to have one platform depend on another platform. Does the sensor platform depend on the climate platform in any other way?

Reply:
The sensor platform has support for actual physical sensors for motion and doors / windows so this part does not.  It also reads heating and hotwater values for providing sensors for current state, current mode, boost state, target temp and current temp and also battery level of the wireless thermostat.

If the min & max values can be added to the climate platform somehow then I am ok with removing current and target temp from the sensor platform, but I think any more info on there and the gui would be too busy for a mobile screen.